<launch>

    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="L_a" type="double" value="0.18" />
    <param name="L_b" type="double" value="0.14" />
    <param name="r_tire" type="double" value="0.05" />

    <!-- control parameters -->
    <param name="input_gain" type="double" value="0.3" />

    <!-- SIMULATOR -->
    <!-- <node pkg="barc" type="vehicle_simulator.py" name="vehicle_simulator" output="screen"/> -->

    <!-- OBSERVER-->
    <node pkg="barc" type="state_estimation_KinBkMdl.py" name="state_estimation" output="screen">
        <!-- v_x estimation sample time-->
        <param name="dt_v_enc" type="double" value="0.2" />

        <!-- ekf properties -->
        <param name="q_std" type="double" value="0.1" />
        <param name="r_std" type="double" value="0.1" />

        <!-- measurement model type -->
        <param name="est_mode" type="int" value="1" />

        <!-- OLD ekf properties
        <param name="q_std" type="double" value="0.01" />
        <param name="psi_std" type="double" value="0.2" />
        <param name="v_std" type="double" value="0.5" />
        <param name="gps_std" type="double" value="0.025" />
        -->
    </node>

    <!-- LOW LEVEL CONTROLLER -->
    <!--<node pkg="barc" type="controller_low_level.py" name="low_level_controller" output="screen"/>-->

    <!-- MPC CONTROLLER -->
    <!--<node pkg="barc" type="controller_MPC_traj.jl" name="MPC" output="screen"/>-->

    <!-- Record the experiment data -->
    <node pkg="rosbag" type="record" name="rosbag_record" args="-o /home/david/simulations/sim_est_KinBkMdl -a" />

    <!-- USE ROSBAG PLAY ... TO REPUBLISH THE RECORDED SENSOR DATA TwentyfiveThree.bag (or other)-->
    <node pkg="rosbag" type="play" name="play_rec_sensor_data" required="true" args="/home/david/simulations/sim_sensor_data/FifteenThreePihalf.bag -d 1" />

</launch>